src/sagemaker/remote_function/job.py

```diff
@@ -860,7 +860,7 @@ def _prepare_and_upload_runtime_scripts(
             )
             shutil.copy2(spark_script_path, bootstrap_scripts)
 
-        with open(entrypoint_script_path, "w") as file:
+        with open(entrypoint_script_path, "w", newline="\n") as file:
```

Comment:
is this change backwards compatible?

Reply:
Yes, it ensures the line endings on the `job_driver.sh` get written out with `\n` as newlines on all platforms instead of `\r\n` on windows.  They need to be `\n` since it will be executed in a linux container.

Currently, it's not an issue if submitting remote functions *from* mac or linux. But submitted them from windows, the script gets written out with `\r\n` the uploaded to s3. Which causes `/opt/ml/input/data/sagemaker_remote_function_bootstrap/job_driver.sh: line 3: $'\r': command not found` when the script is executed on the training image (which we can safely assume will **not** be a windows container).

Side note: windows users run across the line endings on `.sh` scripts quite often.  Another workaround is running something like [`dos2unix`](https://linux.die.net/man/1/dos2unix) on the scripts before executing them on linux, but that's not possible in this case since the sdk is writing them out.